lib/modules/collapseWhitespace.es6

```diff
@@ -35,20 +67,43 @@ export default function collapseWhitespace(tree, options, collapseType, tag) {
 }
 
 
-function collapseRedundantWhitespaces(text, collapseType, isTopLevel = false) {
+function collapseRedundantWhitespaces(text, collapseType, shouldTrim = false, currentTag, prevNodeTag, nextNodeTag) {
     if (!text || text.length === 0) {
         return NONE;
     }
 
     if (collapseType === 'aggressive') {
-        text = text.replace(indentPattern, NONE);
+        text = text.replace(indentPattern, SINGLE_SPACE);
     }
 
     text = text.replace(whitespacePattern, SINGLE_SPACE);
 
-    if (collapseType === 'all' || isTopLevel) {
-        text = text.trim();
+    if (shouldTrim) {
+        if (collapseType === 'aggressive') {
+            if (onlyWhitespacePattern.test(text)) {
```

Comment:
```html
<noTrimWhitespacesArroundElement>text1</noTrimWhitespacesArroundElement>[whitespaces only]<canTrimWhitespacesArroundElement>text2</<canTrimWhitespacesArroundElement>
```

Only the tag before the whitespace is the `noTrimWhitespacesArroundElement` while the after one is not.

If `String#trimEnd` is used, there will be no whitespace left, which will break the page.